phone-book/src/keypad.js

```diff
@@ -0,0 +1,308 @@
+        third: '3'
+    },
+    {
+        first: '4',
+        second: '5',
+        third: '6'
+    },
+    {
+        first: '7',
+        second: '8',
+        third: '9'
+    },
+    {
+        first: '*',
+        second: '0',
+        third: '#'
+    }
+];
+
+class KeypadMode{
+    constructor() {
+        this.keypadModeTemplate = `
+            <div id="keypad-mode" class="contact-block">
+
+                ${this.numberFieldComponent()}
+
+                ${this.keypadComponent()}
+
+                ${this.footerComponent()}
+
+            </div>
+        `
+    }
+
+    numberFieldComponent() {
+        return `
+            <header class="container">
+                <form class="form-inline number-field row">
+                    <div class="form-group">
+                        <span id= "phone-number" class="form-control"></span>
+                        <div id="delete" class="delete-number">
+                        <span class="fas fa-backspace"></span>
+                        </div>
+                    </div>
+                </form>
+            </header>
+        `
+    }
+
+    keypadComponent() {
+        const keypadTemplate = document.createElement('main');
+        keypadTemplate.classList.add('container');
+        keypadTemplate.classList.add('keypad-block');
+
+        const wraperForButtons = document.createElement('div');
+        wraperForButtons.id = 'wraper-for-buttons';
+
+        keypadModeButtons.forEach(buttons => {
+            const rowWithButtons = `
+            <div class="row">
+                <button class="number-btn"><span>${buttons.first}</span></button>
+                <button class="number-btn"><span>${buttons.second}</span></button>
+                <button class="number-btn"><span>${buttons.third}</span></button>
+            </div>
+            `;
+
+            wraperForButtons.innerHTML += rowWithButtons;
+        });
+        
+        keypadTemplate.innerHTML += wraperForButtons.outerHTML;
+
+        const toCallButton = `
+            <div class="row">
+                <div id="call-btn" class="number-btn call"><span class="fas fa-phone"></span></div>
+            </div>
+        `;
+
+        keypadTemplate.innerHTML += toCallButton;
+
+        const errorBlock = `<span id="error-block"></span>`;
+
+        keypadTemplate.innerHTML += errorBlock;
+
+        return keypadTemplate.outerHTML;
+    }
+
+    footerComponent() {
+        return `
+        <footer class="container">
+           
+            <div class="row navigation-panel">
+            
+                <div class="col-4 d-flex justify-content-center">
+                    <a href="contact.html">
+                    <div class="navigation-button">
+                        <span class="far fa-address-book icon"></span>
+                    </div>
+                    </a>
+                </div>
+                
+                <div class="col-4 d-flex justify-content-center">
+                    <a href="#">
```

Comment:
as I see it's a duplication of creating Link. Pretty sure you have to create abstraction over such use case